PracticeGit/Test.c

```diff
@@ -1,13 +1,145 @@
+   if (size1 != size2) return false;
+   for (int i = 0; i < size1; i++)
+      if (arr1[i] != arr2[i]) return false;
+   return true;    // Arrays are equal if all elements match
+}
 
-#include "Program.h"
+void RunQuickSortTest () {
+   int testCases[][MAXSIZE] = {
+       {20, 49, 11, 3, 89, 75, 11, 12, 1},
+       {2, 2, 2, 2, 2, 2, 2},
+       {9},
+       {3, -1, -7, 5, 2, 0, -7},
+       {1, 2, 3, 4, 5, 6},
+       {0, -1, -2, -3, -4, -5},
+       {36, 69},
+       {22, 35, 32, 39, 46, 54, 55},
+       {12, 13, 14, 15, 16, 11},
+   };
+   int expectedResults[][MAXSIZE] = {
+       {1, 3, 11, 11, 12, 20, 49, 75, 89},
+       {2, 2, 2, 2, 2, 2, 2},
+       {9},
+       {-7, -7, -1, 0, 2, 3, 5},
+       {1, 2, 3, 4, 5, 6},
+       {-5, -4, -3, -2, -1, 0},
+       {36, 69},
+       {22, 32, 35, 39, 46, 54, 55},
+       {11, 12, 13, 14, 15, 16}
+   };
+   int sizes[] = { 9, 7, 1, 7, 6, 6, 2, 7, 6 };
+   int elementSearch[] = { 11, 1, 0, -1, 2, 6, 36, 53, 11 };
+   int testCount = sizeof (testCases) / sizeof (testCases[0]);
+   printf ("+--------------------------------------------------+---------------+------------------+--------+\n"
+      "|                Arrays                            | Key to Search | Index of element | Result |\n"
+      "+--------------------------------------------------+---------------+------------------+--------+\n");
+   for (int i = 0; i < testCount; i++) {
+      int size = sizes[i], * arrayToSort = testCases[i];
+      printf ("| Unsorted Array   | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      int searchValue = elementSearch[i];   // Element to search for in the current test case
+      QuickSort (arrayToSort, 0, size - 1);
+      int foundIndex = GetIndex (arrayToSort, size, searchValue);
+      if (foundIndex == ELEMENT_NOT_FOUND) printf ("    %-11d|    Not Found     |", searchValue);
+      else printf ("    %-11d| Found at index %-d |", searchValue, foundIndex);
+      printf ("  %-8s  |\n", AreArraysEqual (arrayToSort, expectedResults[i], size, size) ? GREEN"Pass"RESET : RED"Fail"RESET);
+      printf ("| Sorted Array     | ");
+      PrintArray (arrayToSort, size);
+      printf ("%*s|", (MAXSIZE - size) * 3, " ");
+      printf ("\n+--------------------------------------------------+---------------+------------------+--------+\n");
+   }
+}
+void GetUserInput () {
+   int array[MAXSIZE], size;
+   char input[256], * endptr;
+   while (1) {
+      printf ("Enter the number of elements (1 to %d): ", MAXSIZE);
+      fgets (input, sizeof (input), stdin);
+      size = strtol (input, &endptr, 10);   // Convert input to integer
+      if (*endptr != '\n' && *endptr != '\0') {
+         printf ("Invalid input. Please enter a valid integer.\n");
+         continue;
+      }
+      if (size < 1 || size > MAXSIZE) {
+         printf ("Invalid number of elements. Please enter a value between 1 and %d.\n", MAXSIZE);
+         continue;
```

Comment:
I think it is sufficient to just check valid value and continue as your prompt is informative about input requirements